### PR TITLE
Allow for un-captured route of config.yml

### PR DESCRIPTION
**Pull request by talves**
This fixes if there was an invalid route to the `config.yml` because it did not exist and it was handled without a 404.

Delete this branch when merged.

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 151ae6976241fee59fa49a5471ca24c4e5c8b742

https://deploy-preview-1182--netlify-cms-www.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 151ae6976241fee59fa49a5471ca24c4e5c8b742

https://deploy-preview-1182--cms-demo.netlify.com

**Comment by talves** [comment]
@erquhart This works, but was not sure if I needed to actually parse the empty string config:
`let loadedConfig = parseConfig('');`

File: src/actions/config.js [hunk 1]
```diff
@@ -109,18 +117,16 @@ export function loadConfig() {
     try {
       const preloadedConfig = getState().config;
       const response = await fetch('config.yml', { credentials: 'same-origin' })
-      const requestSuccess = response.status === 200;
+      const fetchedConfig = await getConfigFromResponse(response);
 
-      if (!preloadedConfig && !requestSuccess) {
-        throw new Error(`Failed to load config.yml (${ response.status })`);
+      if (!preloadedConfig && !fetchedConfig) {
+        throw new Error(`Failed to load configuration.`);
```
Comment (erquhart) [inline src/actions/config.js]: The change in error message is to open up to the config no longer being "config.yml" with the understanding that a failed request will already generate a message in the console.

**Comment by erquhart** [comment]
Added a commit for that, let me know what you think.

**Comment by talves** [comment]
The latest commit was what I was proposing to combine the checking of the validity of a yaml download and returning a config based on whether there was a valid file existing and if the preloaded configuration actually exists.

This would also allow for a reload of a configuration later using a different action in the case we end up using dynamic configuration loading. It will also allow for dynamic file path locations to the yaml.

`isPreloaded` defines whether we are going to allow for the config to exist or not, so this should not be bound outside of the getting the config only deciding whether we allow and empty config to return on failure to exist. The validation of the config on merge will determine whether a valid config was loaded.

File: src/actions/config.js [hunk 2]
```diff
@@ -68,6 +68,22 @@ function parseConfig(data) {
...
 }
 
+async function getConfig(file, isPreloaded) {
+  const response = await fetch(file, { credentials: 'same-origin' });
+  if (response.status !== 200) {
+    if (isPreloaded) return parseConfig('');
+    throw new Error(`Failed to load config.yml (${ response.status })`);
+  }
+  const contentType = response.headers.get('Content-Type') || 'Not-Found';
+  const isYaml = contentType.indexOf('yaml') !== -1;
```
Comment (talves) [inline src/actions/config.js]: There are different type strings for yaml that can exist, so this should allow for each.

File: src/actions/config.js [hunk 3]
```diff
@@ -108,19 +124,12 @@ export function loadConfig() {
...
       const preloadedConfig = getState().config;
-      const response = await fetch('config.yml', { credentials: 'same-origin' })
-      const requestSuccess = response.status === 200;
-
-      if (!preloadedConfig && !requestSuccess) {
-        throw new Error(`Failed to load config.yml (${ response.status })`);
-      }
-
-      const loadedConfig = parseConfig(requestSuccess ? await response.text() : '');
+      const loadedConfig = await getConfig('config.yml', preloadedConfig && preloadedConfig.size > 1);
```
Comment (talves) [inline src/actions/config.js]: In testing, preloadedConfig has an iterable object map with one key `isFetching` if the preload did not have a JSON preload using `init()`.

**Comment by talves** [comment]
@erquhart did you forget to merge this prior to #1173 ?

**Pull request merged**